tests/test_dark_magic.py

```diff
@@ -581,3 +590,48 @@ class FooError(Exception):
             x = attr.ib()
 
         FooError(1)
+
+    @pytest.mark.parametrize("slots", [True, False])
+    @pytest.mark.parametrize("frozen", [True, False])
+    def test_eq_only(self, slots, frozen):
+        """
+        Classes with order=False cannot be ordered.
+
+        Python 3 throws a TypeError, in Python2 we have to check for the
+        absence.
+        """
+
+        @attr.s(eq=True, order=False, slots=slots, frozen=frozen)
+        class C(object):
+            x = attr.ib()
+
+        if not PY2:
+            possible_errors = (
+                "unorderable types: C() < C()",
+                "'<' not supported between instances of 'C' and 'C'",
+                "unorderable types: C < C",  # old PyPy 3
+            )
+
+            with pytest.raises(TypeError) as ei:
+                C(5) < C(6)
+
+            assert ei.value.args[0] in possible_errors
+        else:
+            i = C(42)
+            for m in ("lt", "le", "gt", "ge"):
+                assert None is getattr(i, m, None)
```

Comment:
This looks broken (passing always accidentally).

Probably should use the same kind of test as above where you actually run comparison instead of testing negatives.

Reply:
How? Python 2 happily compares incomparable types.

Reply:
Possibly by just checking you get the default behavior. Either by creating two classes and comparing them and seeing it doesn't blow up even if you compare them to each other, or by creating two instances and asserting the one with higher ID won.

But if all that sounds like too much work, glad at least that it looks like you added the double underscores so that this does something now :)

Reply:
It does, let's keep it that way. :)